crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -720,6 +720,154 @@ impl<TX: DbTxMut + DbTx> DatabaseProvider<TX> {
         ))
     }
 
+    /// Remove the last N blocks of state, recreating the [`ExecutionOutcome`].
```

Comment:
should it be just
```suggestion
    /// Remove the last N blocks of state.
```
because we don't return anything?